### PR TITLE
app: improved notifications UI

### PR DESCRIPTION
**Pull request by buck54321**
The note bell gets a number instead of a dot. All notifications > POKE now show in the poke notes (renamed to popup notes). POKE-severity notification history is now available as through the bell dropdown as a "Recent Activity" feed, but is not persistent across sessions. 

Up to 100 notifications will be shown in the bell dropdown now (used to be 10).

The popup notifications can now be disabled on the settings page. Default is on.

Settings page is given a proper layout.

A handful of other CSS tweaks.

### CODE REVIEW

File: client/webserver/middleware.go [hunk 1]
```diff
@@ -37,28 +37,31 @@ func securityMiddleware(next http.Handler) http.Handler {
...
+	cookie, err := r.Cookie(k)
+	switch err {
+	// Dark mode is the default
+	case nil:
+		return cookie.Value == "1"
+	case http.ErrNoCookie:
+	default:
+		log.Errorf("Cookie %q retrieval error: %v", k, err)
+	}
+	return deFault
```
Comment (chappjc) [inline client/webserver/middleware.go:62]: de**F**ault

File: client/webserver/site/src/js/app.js [hunk 2]
```diff
@@ -470,33 +536,51 @@ export default class Application {
...
   }
 
-  /* setNoteElements re-builds the drop-down notification list. */
-  setNoteElements () {
+  prependPokeElement (note) {
+    this.pokes.push(note)
+    while (this.pokes.length > noteCacheSize) this.pokes.shift()
+    const el = this.makePoke(note)
+    const noteList = this.page.pokeList
+    this.prependListElement(noteList, note, el)
```
Comment (chappjc) [inline client/webserver/site/src/js/app.js:544]: Need the `noteList` variable?

File: client/webserver/site/src/js/doc.js [hunk 3]
```diff
@@ -102,6 +102,11 @@ export default class Doc {
     return el.classList.contains('d-hide')
   }
 
+  /* isDisplayed returns true if the specified element is not hidden */
+  static isDisplayed (el) {
+    return !el.classList.contains('d-hide')
```
Comment (chappjc) [inline client/webserver/site/src/js/doc.js:107]: Wouldn't it be tighter to call `!isHidden` instead of defining `isDisplayed`?

**Review by chappjc — CHANGES_REQUESTED** [review]
Looks good, works well.  Just a few nits.

File: client/webserver/site/src/js/doc.js (hunk 3 above)
Comment (buck54321) [reply client/webserver/site/src/js/doc.js:107]: The two are equivalent, but I wasn't crazy about `!isHidden` where I needed it. I thought `isDisplayed` reads better in some contexts.

**Review by chappjc — APPROVED** [review]

**Pull request merged**